cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/AbstractInMemoryProgramRunner.java

```diff
@@ -80,7 +80,8 @@ public AbstractInMemoryProgramRunner(CConfiguration cConf, ProgramRunnerFactory
   private void startComponent(Program program, String name, int instances, RunId runId, ProgramOptions options,
                                 Table<String, Integer, ProgramController> components, ProgramRunnerFactory.Type type) {
     for (int instanceId = 0; instanceId < instances; instanceId++) {
-      ProgramOptions componentOptions = createComponentOptions(name, instanceId, instances, runId, options);
+      ProgramOptions componentOptions = createComponentOptions(name + "-" + instanceId,
```

Comment:
isn't that redundant? So if I have a flowlet named "parser", then the name will be "parser-0" and the instance id "0" etc?